src/librustc_borrowck/borrowck/mir/mod.rs

```diff
@@ -89,3 +192,127 @@ impl<'b, 'a: 'b, 'tcx: 'a> MirBorrowckCtxt<'b, 'a, 'tcx> {
         debug!("MirBorrowckCtxt::process_terminator({:?}, {:?})", bb, term);
     }
 }
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+enum DropFlagState {
+    Live,
+    Dead
```

Comment:
I made this comment elsewhere, but I think we should rename these to `Initialized` and `Deinitialized` to avoid confusion about "live vs dead" from liveness analysis (though I suppose that if the type impls drop --or we aren't sure, in which case there is a Drop(x) anyway -- there is no difference...so maybe I am wrong and we should leave it as is.)